sdk/trace/tracetest/recorder.go

```diff
@@ -69,6 +69,19 @@ func (sr *SpanRecorder) Started() []sdktrace.ReadWriteSpan {
 	return dst
 }
 
+// Reset clears the recorded spans.
+//
+// This method is safe to be called concurrently.
+func (sr *SpanRecorder) Reset() {
+	sr.startedMu.Lock()
+	sr.started = nil
+	sr.startedMu.Unlock()
+
+	sr.endedMu.Lock()
+	sr.ended = nil
+	sr.endedMu.Unlock()
```

Comment:
It's unlikely to happen, but imagine the following:

* Reset is called
* The `startedMu` mutex is locked, the data is reset
* The `startedMu` mutex is unlocked
* A new span is started
* The `ended` mutex is locked

We'd end up with a race condition where despite resetting, we have a span at the end of the call.
How about locking both mutexes before clearing the data?

Reply:
I agree with @dmathieu. The only important thing is to lock in the same order if there will be other places which need to use both locks. Otherwise we can have a deadlock.